lib/usd/translators/mayaReferenceUpdater.cpp

```diff
@@ -108,6 +112,49 @@ bool PxrUsdTranslators_MayaReferenceUpdater::shouldAutoEdit() const
 /* virtual */
 bool PxrUsdTranslators_MayaReferenceUpdater::canEditAsMaya() const { return true; }
 
+/* virtual */
+bool PxrUsdTranslators_MayaReferenceUpdater::editAsMaya()
+{
+    // Lock the transform node that corresponds to the Maya reference prim.
+    MDagPath transformPath;
+    if (MDagPath::getAPathTo(getMayaObject(), transformPath) != MS::kSuccess) {
+        return false;
+    }
+
+    MayaUsd::LockNodesUndoItem::lock(
+        "Maya reference pulled transform locking", transformPath, true);
+
+    // Lock all attributes except the transform attributes.  Create a set of
+    // Maya attributes that are converted to USD transform attributes.
+    // Children of compounds are ignored, as their parent will be locked.
+    static std::set<std::string> xformAttrNames;
+    if (xformAttrNames.empty()) {
+        for (const auto& opClass : UsdMayaXformStack::MayaStack().GetOps()) {
+            if (!opClass.IsInvertedTwin()) {
+                xformAttrNames.insert(std::string(opClass.GetName().GetText()));
+            }
+        }
+    }
+
+    MFnDependencyNode fn(getMayaObject());
```

Comment:
Core change: lock all attributes except transform attributes.  Re-used dummy transform creation algorithm to determine what is considered a transform attribute.